modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/WorkspaceExtension.java

```diff
@@ -365,6 +375,53 @@ private String _decodeBundleUrl(ProductInfo productInfo) {
 		}
 	}
 
+	private String _getBundleChecksumMD5() {
+		String bundleUrl = getBundleUrl();
+
+		if (Objects.isNull(bundleUrl) || bundleUrl.isEmpty()) {
+			return null;
+		}
+
+		try {
+			DownloadCommand downloadCommand = new DownloadCommand();
+
+			downloadCommand.setCacheDir(_workspaceCacheDir);
+			downloadCommand.setPassword(null);
+			downloadCommand.setToken(false);
+			downloadCommand.setUrl(new URL(bundleUrl + ".MD5"));
+			downloadCommand.setUserName(null);
+			downloadCommand.setQuiet(true);
+
+			downloadCommand.execute();
+
+			Path md5Path = downloadCommand.getDownloadPath();
+
+			String md5Content = FileUtil.read(md5Path.toFile());
+
+			int lastSlash = bundleUrl.lastIndexOf('/');
```

Comment:
you should check lastSlash before you do the download.  It doesn't make sense to do the download is lastSlash == -1.

Reply:
Why would last slash not be there? what test case are you thinking about?